### ISSUE TITLE
Update schema files in OPTIMADE main repository before releasing v1.0.0-rc.2

### ISSUE THREAD

**Issue by rartino** [Materials-Consortia/OPTIMADE#286]
All PRs that we said to merge before release of v1.0.0-rc.2 have now been merged, and I am ready to make that release.

However, just now I realize that some of the changes we've made should have affected the schema files:
- schemas/index_openapi_schema.json
- schemas/openapi_schema.json
which, as far as I know, are not yet updated?

Do we not need to update them before we can release?

**Comment by rartino** [comment]
Pinging @CasperWA, @ml-evs who I know are active in the optimade-python-tools that presently is used to generate the schemas.

**Comment by ml-evs** [comment]
> Pinging @CasperWA, @ml-evs who I know are active in the optimade-python-tools that presently is used to generate the schemas.

We have a few open issues in optimade-python-tools for the models to be updated; I'll assign myself and start chipping away at them tomorrow, with some help it shouldn't take long. Would you be against releasing v1.0.0-rc2 now and then moving the tag back once we've caught up?

**Comment by rartino** [comment]
> Would you be against releasing v1.0.0-rc2 now and then moving the tag back once we've caught up?

I am against making a release with incorrect schemas for that release since that is a bad look..., but I realize this may further put off -rc.2 for some time. (But I am not sure what "moving the tag back" means? Altering the git tag for the release after the release? That we shouldn't do.)

While we are on the topic of schemas and the v1.0.0 release, I notice that both #25 (Adoption of json schemas and continous integration) and #37 (API response should include reference to schema, to be completed first) are still open and have been marked for milestone v1.0.

- I've proposed to close #25, because I think it is done. 
- For #37, I suppose we do not presently link schemas from our default JSON API-based response? Should we quick-fix that before v1.0.0? At least an optional way to link schemas may be appreciated for v1.0.0 and isn't a big change to the spec.

**Comment by ml-evs** [comment]
> I am against making a release with incorrect schemas for that release since that is a bad look..., but I realize this may further put off -rc.2 for some time. (But I am not sure what "moving the tag back" means? Altering the git tag for the release after the release? That we shouldn't do.)

Yes, it would be a bit cheeky, and relies on the fact that providers probably aren't using the generated OpenAPI spec at the moment. I guess the problem is that all other PRs would need to be put on hold until the schemas are merged.

We can certainly try our best to expedite the process in optimade-python-tools!

>     * For #37, I suppose we do not presently link schemas from our default JSON API-based response? Should we quick-fix that before v1.0.0? At least an optional way to link schemas may be appreciated for v1.0.0 and isn't a big change to the spec.

This sounds good to me, I would like to be able to link to my extended OpenAPI OPTIMADE schema that includes the extra fields for my implementation (which itself will link to the OPTIMADE schema).

**Comment by rartino** [comment]
@ml-evs, @CasperWA 

I see you actively working on it in optimade-python-tools, how much more work is left? Are there clear work tasks others can help you with to get this completed?

**Comment by ml-evs** [comment]
> @ml-evs, @CasperWA
> 
> I see you actively working on it in optimade-python-tools, how much more work is left? Are there clear work tasks others can help you with to get this completed?

Hi @rartino, just one more PR to be reviewed at our end, https://github.com/Materials-Consortia/optimade-python-tools/pull/351,  unfortunately its a tedious one as it is just the update of all our descriptions to match the spec (and also converted to markdown so our docs work). As soon as this is merged, I can prepare the PR for adding the generated OpenAPI spec to this repo.

**Comment by ml-evs** [comment]
Closed by #291.
